node/index.js

```diff
@@ -78,24 +78,33 @@ function jsonGet(options, callback) {
   }
 */
 app.get('/announce', function (req, res) {
-    var shutdown = req.query.shutdown !== undefined ? (req.query.shutdown === "true" || req.query.shutdown === "1") : false;
-    var serverPort = req.query.port || 0;
-    if (serverPort === 0) {
+    var shutdown = req.query.shutdown === "true" || req.query.shutdown == 1;
+    
+    if (!req.query.port) {
         return res.send({result: {code: 1, msg: "Invalid parameters, valid parameters are 'port' (int) and 'shutdown' (bool)"}});
     }
+    var serverPort = +req.query.port;
+    if (isNaN(serverPort) || serverPort < 1024 || serverPort > 65535) {
+      return res.send({result: {code: 4, msg: "Invalid port. A valid port is in the range 1024-65535."}}); //could allow 1-65535
+    }
 
     var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress;
```

Comment:
The port? Sure, someone can give it any port, but the point is to ensure that the supplied port is an integer in a valid range. We really just don't want someone like me adding "?callback=?" to the end of the port, potentially forcing jQuery to get the server information as JSONP.